custom_components/thermal_comfort/sensor.py

```diff
@@ -237,27 +243,73 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     return True
 
 
-class SensorThermalComfort(SensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    sensor_types: list[SensorType] = []
+    entity_registry = er.async_get(hass)
+    _LOGGER.debug(f"async_setup_entry: {data}")
+    for c in SensorType:
+        if data.get(c, False):
+            sensor_types.append(c)
+        else:
+            eid = entity_registry.async_get_entity_id(
+                domain="sensor",
+                platform=DOMAIN,
+                unique_id=id_generator(config_entry.unique_id, c)
+            )
+            if eid is not None:
+                entity_registry.async_remove(eid)
+            else:
+                _LOGGER.debug(f"async_setup_entry: should remove sensor {c}, but entity_id from registry is None")
+
+    compute_device = DeviceThermalComfort(
+        hass,
+        data[CONF_TEMPERATURE_SENSOR],
+        data[CONF_HUMIDITY_SENSOR],
+        sensor_types,
+        data[CONF_POLL],
+    )
+
+    entities: list[SensorThermalComfortEntry] = []
+    for sensor_type in sensor_types:
+        _LOGGER.debug(f"async_setup_entry: creating entity for {sensor_type}")
+        entity_description = SensorEntityDescription(**SENSOR_TYPES[sensor_type])
+        entities.append(
+            SensorThermalComfortEntry(
+                device=compute_device,
+                entity_description=entity_description,
+                sensor_type=sensor_type,
+                device_name=data[CONF_NAME],
+                device_id=f"{config_entry.unique_id}",
```

Comment:
we shouldn't use the unique_id here

Reply:
Why? It goes to:
1. `_attr_device_info` as part of identifier (like in [yeelight](https://github.com/home-assistant/core/blob/881d35ab171007696f819d3a3e84499d0cbf6728/homeassistant/components/yeelight/entity.py#L20) or [nut](https://github.com/home-assistant/core/blob/80b619550592240a8d15c0524b48f595ff265885/homeassistant/components/nut/sensor.py#L81));
2. `SensorThermalComfortCommon` as `unique_id`, like code before.

Reply:
The device_id in the previous code uses the identifier of the sensor config entry in the yaml.
Resulting in `sensor.{identifier}_{sensory_type}` and not sensor.{unique_id}_{sensor_type}.
Imagine some random number unique_id in your entity id. Not so great to use or readable.
`unique_id != entity_id`

Reply:
So we may use `config_entry.unique_id` for identifiers (because it is unique identifier), but for sensors entry_id we should use `config_entry.entry_id`, because it looks pretty. Right?

Reply:
Yes and no. Not because it looks pretty, but because it has to be a user readable string.
Unique id is the system internal unique identifier not visible to the user. Entity id is the identifier a human sees and uses to interact with the entity all over the frontend.

Reply:
I see.
Will update this part.

Reply:
`config_entry.entry_id` is look like `d321966f95ba46a8cdfde7d86509c451` and we should not use it too.
`data[CONF_NAME]` may be the best choose.
And parameter in constructor should have clear name.

Reply:
> And parameter in constructor should have clear name.

Yes your PR comes in shortly after a refactor where i didn't want to change to much of the previous names.

> data[CONF_NAME] may be the best choose

Yes. This is also a decision we have to make for the new yaml config and that is name handling and how it interacts with the entity id. Legacy yaml has the `identifier` and `friendly_name`. The original plan was to move `identifier` and add it as `name` entry. The question is if we should reduce this to `friendly_name` require it and generate the `enity_id` from it and to the same here with data[CONF_Name].

But that would mean we have to either make `friendly_name` a required entry or have to generate one if `friendly_name` isn't set. e.g. something like you did for the entity_id right now e.g. combining the entity_id's of the input sensors.

Reply:
For configuration from user interface friendly name is asked from user. Then it goes to device name in UI. If I change device name in UI, then HA offer to update sensors according to new name.